webapp/components/TeaserImage/TeaserImage.vue

```diff
@@ -113,29 +119,41 @@ export default {
     },
     cropImage() {
       this.showCropper = false
+      if (this.file.type === 'image/jpeg') {
+        this.uploadJpeg()
+      } else {
+        this.uploadOtherImageType()
+      }
+    },
+    uploadOtherImageType() {
+      this.imageAspectRatio = this.file.width / this.file.height || 1.0
+      this.image = new Image()
+      this.image.src = this.file.dataURL
+      this.setupPreview()
+      this.emitImageData(this.file)
+    },
+    uploadJpeg() {
       const canvas = this.cropper.getCroppedCanvas()
       canvas.toBlob(blob => {
-        const imageAspectRatio = canvas.width / canvas.height
-        this.setupPreview(canvas)
-        this.removeCropper()
+        this.imageAspectRatio = canvas.width / canvas.height
+        this.image = new Image()
+        this.image.src = canvas.toDataURL()
+        this.setupPreview()
         const croppedImageFile = new File([blob], this.file.name, { type: this.file.type })
-        this.$emit('addTeaserImage', croppedImageFile)
-        this.$emit('addImageAspectRatio', imageAspectRatio)
+        this.emitImageData(croppedImageFile)
       }, 'image/jpeg')
     },
-    setupPreview(canvas) {
-      this.image = new Image()
-      this.image.src = canvas.toDataURL()
+    setupPreview() {
       this.image.classList.add('thumbnail-preview')
       this.thumbnailElement.appendChild(this.image)
     },
     cancelCrop() {
-      this.showCropper = false
       if (this.oldImage) this.thumbnailElement.appendChild(this.oldImage)
-      this.removeCropper()
+      this.showCropper = false
     },
-    removeCropper() {
-      this.editor.removeChild(document.querySelectorAll('.cropper-container')[0])
+    emitImageData(imageFile) {
+      this.$emit('addTeaserImage', imageFile)
+      this.$emit('addImageAspectRatio', this.imageAspectRatio)
```

Comment:
This adds a lot of complexity! @ogerly could we maybe do a pair-programming together. I have the same feeling towards that code as I had lately with the reports.

Reply:
we agreed that we would leave this for now, but @alina-beck if you have a better way to achieve this, we would be happy to avoid emitting this three times and having additional `state` for `isCropInProgress` on the `ContributionForm`